File: ReactRedux/src/models/EnumMetadata.ts
import { EnumValueMetadata, initEnumValueMetadata } from "./EnumValueMetadata";
import { ProjectMetadata, initProjectMetadata } from "./ProjectMetadata";
export interface EnumMetadata {
  idEnumMetadata:number;
  name:string;
  caption:string;
  values:EnumValueMetadata[];
  idProjectMetadata:number;
  projectMetadata?:ProjectMetadata;
}

export const initEnumMetadata = {
  idEnumMetadata: 0,
  name: '',
  caption: '',
  values: [],
  idProjectMetadata: 0,

}

File: DataAccessLayer/Models/EnumMetadata.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DataAccessLayer.Dto
{
    public class EnumMetadata
    {

        public EnumMetadata()
        {
          Values = new HashSet<EnumValueMetadata>();

        }

        [Key]
        public int IdEnumMetadata { get; set; }
        public string Name { get; set; }
        public string Caption { get; set; }

        
        public virtual ICollection<EnumValueMetadata>? Values { get; set; }
        public int? IdProjectMetadata { get; set; }

        
        [ForeignKey("IdProjectMetadata")]
        public virtual ProjectMetadata? ProjectMetadata { get; set; }
    }
}
